### ISSUE TITLE
Support `string.size()` alongside `string.length()`

### ISSUE THREAD

**Issue by Arecher** [godotengine/godot-proposals#7869]
### Describe the project you are working on

Game project in Godot 3.5

### Describe the problem or limitation you are having in your project

From time to time, especially when looping over a `string` as an `Array`, I automatically write `string.size()` rather than `string.length()`, which doesn't exist.
```GDscript
for characterposition in string.size():
    pass
```
At first I thought of this as a silly mistake on my end, but when I thought about it, I concluded that 'string.size()' actually makes perfect sense. The documentation states that Strings are `An Array of Characters`, and they handle as an array in a lot of use-cases, such as:

```GDscript
for character in "string":
    print(character)
# prints "s" "t" "r" "i" "n" "g"
```
or

```GDscript
var thirdcharacter = "string"[2]
print(thirdcharacter)
# prints "r"
```
Which makes it feel strange that they don't support getting the size of the "array" as you'd expect. 

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Add support for `string.size()` alongside `string.length()`. They will both return the exact same value.

It makes sense that strings have their own way to get the length, since this is a lot easier to understand for new users that might not use/need to use strings as arrays a lot. But for users that do strings as arrays, supporting them in small ways such as allowing `string.size()` would make the Engine feel smarter and more consistent in it's usage.

Now I understand this will get the _"syntactic sugar"_ comments right away, and I know you want to keep the codebase as small as possible. But small additions like this, especially when they make sense, can help the user experience. And it's not like making `string.size()` internally call `string.length()` suddenly creates a lot of new code to maintain. Just because something can already be done in one way, doesn't mean another way cannot be just as valid!

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

See above.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

You can always use 'length()'. 

### Is there a reason why this should be core and not an add-on in the asset library?

String is core.

**Comment by mieldepoche** [comment]
looking a bit at the docs:
- `size` often refers to either a number of elements (`int`, `Vector2i`: array, texture size, font size) or a spatial size (`float`, `Vector2` or `Vector3`: mesh or collision shape extends, etc.).
- `length` almost always refers to a spatial or temporal length (`float`: the length of an audio effect buffer in seconds, of a bone, of a vector2/3, ...)

So I think it would make sense to rename it.
Alongside with `FileAcess.get_length()` which funnily enough says it returns the `size` of the file
![image](https://github.com/godotengine/godot-proposals/assets/71937306/687b282a-b547-49ad-a514-c2e73b53d629)

**Comment by Zireael07** [comment]
> So I think it would make sense to rename it.

The proposal is not to rename, is to make a second option. A straight up rename is impossible for compatibility reasons

**Comment by Mickeon** [comment]
There's almost no precedents for aliases among the methods in Godot. This is because they risk making the API harder to understand.

One interesting exception is Array's `push_back` which is perfectly identical to `append`, but exists solely to complement `push_front` and other similarly named methods.

There can be a pretty strong argument for `String.size()`, thoigh, as the proposal entails. I myself had to relearn to use "length" at first, because other languages may simply call it "size", instead.
Calling it "length" makes sense to me now, when thinking of it as a series of characters, not in the programming sense.

Do note that, while Strings are essentially arrays of characters, they do not behave like Godot's Arrays internally. "_Copy-on-write_" means that a **String** is not able to change its length, because every change to it returns a whole new String. But, an array can change its size, anytime, and still be the same reference.

-------

##### If in doubt, you may also use the GDScript method `len()`, by the way.

**Comment by AThousandShips** [comment]
Internally `String` has `length` and `size`, where they differ in value, `size` includes the termination and `length` does not, the exact meaning of the two is a complex thing

**Comment by ewrogers** [comment]
I'm not sure how Godot handles strings internally but this is one issue that becomes tricky with encodings. Can non-Latin characters be stored? CJK characters? Other scripts with complex modifiers? Emojis?

What do you expect `.length()` to return for the string "👋🏻 Hello World 🌎!"? What would `.size()` return, and it would always be identical?

Strings are very complex unless you stick with ASCII.

**Comment by Mickeon** [comment]
> Can non-Latin characters be stored? CJK characters? Other scripts with complex modifiers? Emojis?

Yes to all. As by the Unicode standard. Some characters may not be rendered as expected, but they are stored regardless.

> What do you expect `.length()` to return for the string "👋🏻 Hello World 🌎!"? What would `.size()` return, and it would always be identical?

In a programming context, and as the proposal suggests, they should be identical. An emoji ZWJ sequence is composed by multiple characters, and they should be treated as such, ideally.

**Comment by MewPurPur** [comment]
@AThousandShips Is this saying that String.size() in the codebase equals String.length() + 1?

**Comment by Volutar** [comment]
Isn't it related to utf8 string measurements? String can be of 18 bytes long, but only 10 characters (simply because characters are not fixed-sized)
